client/my-sites/checkout/composite-checkout/components/jetpack-pro-redirect-modal/index.tsx

```diff
@@ -49,6 +50,24 @@ export default function JetpackProRedirectModal( { redirectTo, productSourceFrom
 
 	const isAgencyPartner = useSelector( isAgencyUser );
 
+	// Function to record the event when the modal is displayed.
+	// It is in a separate useEffect to avoid unecessary re-renders.
+	useEffect( () => {
+		const recordShowEvent = () => {
+			dispatch( recordTracksEvent( 'jetpack_dashboard_agency_checkout_redirect_modal_show' ) );
+		};
+
+		if (
+			isAgencyPartner &&
+			! isDismissed &&
+			( productSourceFromUrl === 'jetpack-plans' ||
+				redirectURLPage === 'my-jetpack' ||
```

Comment:
How about we create a variable to check if the source is Jetpack to avoid repeating these lines. We could also use it here: https://github.com/Automattic/wp-calypso/pull/80223/files#diff-ec135213b0b18228ef74726244533ef8900a9536b66f02e380432377740b6bcdR77

```
const isJetpackSource =
		productSourceFromUrl === 'jetpack-plans' ||
		redirectURLPage === 'my-jetpack' ||
		redirectURLPage === 'jetpack';
```

Reply:
Great suggestion :)  Done!